src/common/legacy_config_opts.h

```diff
@@ -945,143 +924,9 @@ OPTION(bdev_nvme_retry_count, OPT_INT) // -1 means by default which is 4
 
 OPTION(objectstore_blackhole, OPT_BOOL)
 
-OPTION(bluefs_alloc_size, OPT_U64)
-OPTION(bluefs_max_prefetch, OPT_U64)
-OPTION(bluefs_min_log_runway, OPT_U64)  // alloc when we get this low
-OPTION(bluefs_max_log_runway, OPT_U64)  // alloc this much at a time
-OPTION(bluefs_log_compact_min_ratio, OPT_FLOAT)      // before we consider
-OPTION(bluefs_log_compact_min_size, OPT_U64)  // before we consider
-OPTION(bluefs_min_flush_size, OPT_U64)  // ignore flush until its this big
-OPTION(bluefs_compact_log_sync, OPT_BOOL)  // sync or async log compaction?
-OPTION(bluefs_buffered_io, OPT_BOOL)
-OPTION(bluefs_sync_write, OPT_BOOL)
-OPTION(bluefs_allocator, OPT_STR)     // stupid | bitmap
-OPTION(bluefs_preextend_wal_files, OPT_BOOL)  // this *requires* that rocksdb has recycling enabled
-
-OPTION(bluestore_bluefs, OPT_BOOL)
-OPTION(bluestore_bluefs_env_mirror, OPT_BOOL) // mirror to normal Env for debug
-OPTION(bluestore_bluefs_min, OPT_U64) // 1gb
-OPTION(bluestore_bluefs_min_ratio, OPT_FLOAT)  // min fs free / total free
-OPTION(bluestore_bluefs_max_ratio, OPT_FLOAT)  // max fs free / total free
-OPTION(bluestore_bluefs_gift_ratio, OPT_FLOAT) // how much to add at a time
-OPTION(bluestore_bluefs_reclaim_ratio, OPT_FLOAT) // how much to reclaim at a time
-OPTION(bluestore_bluefs_balance_interval, OPT_FLOAT) // how often (sec) to balance free space between bluefs and bluestore
-// If you want to use spdk driver, you need to specify NVMe serial number here
-// with "spdk:" prefix.
-// Users can use 'lspci -vvv -d 8086:0953 | grep "Device Serial Number"' to
-// get the serial number of Intel(R) Fultondale NVMe controllers.
-// Example:
-// bluestore_block_path = spdk:55cd2e404bd73932
-OPTION(bluestore_block_path, OPT_STR)
-OPTION(bluestore_block_size, OPT_U64)  // 10gb for testing
-OPTION(bluestore_block_create, OPT_BOOL)
-OPTION(bluestore_block_db_path, OPT_STR)
-OPTION(bluestore_block_db_size, OPT_U64)   // rocksdb ssts (hot/warm)
-OPTION(bluestore_block_db_create, OPT_BOOL)
-OPTION(bluestore_block_wal_path, OPT_STR)
-OPTION(bluestore_block_wal_size, OPT_U64) // rocksdb wal
-OPTION(bluestore_block_wal_create, OPT_BOOL)
-OPTION(bluestore_block_preallocate_file, OPT_BOOL) //whether preallocate space if block/db_path/wal_path is file rather that block device.
-OPTION(bluestore_csum_type, OPT_STR) // none|xxhash32|xxhash64|crc32c|crc32c_16|crc32c_8
-OPTION(bluestore_min_alloc_size, OPT_U32)
-OPTION(bluestore_min_alloc_size_hdd, OPT_U32)
-OPTION(bluestore_min_alloc_size_ssd, OPT_U32)
-OPTION(bluestore_max_alloc_size, OPT_U32)
-OPTION(bluestore_prefer_deferred_size, OPT_U32)
-OPTION(bluestore_prefer_deferred_size_hdd, OPT_U32)
-OPTION(bluestore_prefer_deferred_size_ssd, OPT_U32)
-OPTION(bluestore_compression_mode, OPT_STR)  // force|aggressive|passive|none
```

Comment:
could you add `.set_enum_allowed({"force", "aggressive", "passive", "none"})` for `bluestore_compression_mode`?